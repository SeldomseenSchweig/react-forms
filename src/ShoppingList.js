import { randomUUID } from 'crypto';
import React, {useState} from 'react';
import Item from './item';
import NewItemForm from './NewItemForm';
import { v4 as uuid } from 'uuid'




const ShoppingList =()=>{
    const INITIAL_STATE = [
        {id:uuid(),name:"AB", quantity:3},
        {id:uuid(),name:"Apple", quantity:5}
    ]
    const [items, setItems] = useState(INITIAL_STATE)
    const addItem = (newItem)=>{
        setItems(items =>([...items,{...newItemid, id:uuid()} ]))
    }

    return (

        <div>
            <NewItemForm addItem={addItem} />
            <h1> Shopping List</h1>
            <div>
            
            { items.map(({id,name,quantity}) =><Item id={id} name={name} quantity={quantity} key={id}/> )}
            </div>

            
        </div>
    )



}

export default ShoppingList;