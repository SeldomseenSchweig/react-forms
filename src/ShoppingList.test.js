import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ShoppingList from "./ShoppingList";
import '@testing-library/jest-dom/extend-expect';


test("renders without crashing", function (){

    render(<ShoppingList />);
});

test("matches snapshot", function () {
    const {asFragment} = render(<ShoppingList />)
    expect (asFragment()).toMatchSnapshot();
});

test("should add new item", ()=>{

    const {queryByText, getByLabelText} = render(<ShoppingList />)
    const input = getByLabelText('Product Name')
    const button = queryByText('Submit')
    expect(queryByText('Product Name: Peanut Butter')).not.toBeInTheDocument();
    fireEvent.change(input,{target:{value: "Peanut Butter"}});
    fireEvent.click(button);
    expect(queryByText('Product Name: Peanut Butter')).toBeInTheDocument();




})