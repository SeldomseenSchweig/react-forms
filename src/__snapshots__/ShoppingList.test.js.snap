// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="name"
      >
         Product Name
      </label>
      <input
        id="name"
        name="name"
        placeholder="Item Name"
        type="text"
        value=""
      />
      <label
        for="quantity"
      >
         Quantity 
      </label>
      <input
        id="quantity"
        max="10"
        min="1"
        name="quantity"
        type="range"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <h1>
       Shopping List
    </h1>
    <div>
      <ul>
        <li>
          Product Name: AB
        </li>
        <li>
          Quantity: 3
        </li>
      </ul>
      <ul>
        <li>
          Product Name: Apple
        </li>
        <li>
          Quantity: 5
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
